apps/etterlatte-beregning/src/main/kotlin/beregning/grunnlag/GrunnlagMedPeriode.kt

```diff
@@ -0,0 +1,128 @@
+package no.nav.etterlatte.beregning.grunnlag
+
+import no.nav.etterlatte.libs.regler.PeriodisertGrunnlag
+import java.time.LocalDate
+
+data class GrunnlagMedPeriode<T>(
+    val data: T,
+    val fom: LocalDate,
+    val tom: LocalDate? = null
+) {
+    init {
+        if (tom != null && fom > tom) {
+            throw UgyldigPeriodeForGrunnlag(fom, tom)
+        }
+    }
+}
+
+class UgyldigPeriodeForGrunnlag(fom: LocalDate, tom: LocalDate?) :
+    Exception("En periode kan ikke ha fra og med ($fom) etter til og med ($tom)!")
+
+fun <T, R> List<GrunnlagMedPeriode<T>>.mapVerdier(mapVerdi: (T) -> R): List<GrunnlagMedPeriode<R>> {
+    return this.map { GrunnlagMedPeriode(data = mapVerdi(it.data), fom = it.fom, tom = it.tom) }
+}
+
+private val kastFeilUtenforPerioder =
+    { dato: LocalDate, _: LocalDate, _: LocalDate? -> throw PeriodiseringAvGrunnlagFeil.DatoUtenforPerioder(dato) }
+
+object PeriodisertBeregningGrunnlag {
+
+    private class Grunnlag<T>(
```

Comment:
Trenger vi en test for denne?

Reply:
Den testes implisitt via bruken av funksjonene for å få laget et `Grunnlag`. Siden `Grunnlag` ikke er `public` holder det å teste garantiene gitt til funksjonene som lager `Grunnlag`, men mulig du er uenig her? Jeg vil bare unngå å gjøre `Grunnlag` public